BlueDBOnDisk/src/main/java/org/bluedb/disk/BlueDbOnDisk.java

```diff
@@ -21,6 +22,7 @@
 
 	private final Path path;
 	private final BackupManager backupManager;
+	private final Timer sharedTimer = new Timer();
```

Comment:
Do you know what happens if one of your tasks throws an uncaught exception/error? Will it kill the timer thread? Will it stop that task from running again? You might want to put try catches on all your tasks. ScheduledThreadPoolExecutors are newer and the java doc is more specific with what happens in those cases.

Reply:
sounds reasonable to change to ScheduledThreadPoolExecutor